File: frontend/src/types/services.ts
export interface Service {
  id: string;
  user_id: string;
  name: string;
  description?: string;
  duration_minutes: number;
  price: number;
  created_at: string;
  updated_at: string;
}

export interface ServiceCreate {
  name: string;
  description?: string;
  duration_minutes: number;
  price: number;
}

export interface ServiceUpdate {
  name?: string;
  description?: string;
  duration_minutes?: number;
  price?: number;
}

File: backend/app/schemas/services.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from decimal import Decimal


class ServiceBase(BaseModel):
    name: str
    description: Optional[str] = None
    duration_minutes: int = 60
    price: Decimal


class ServiceCreate(ServiceBase):
    pass


class ServiceUpdate(ServiceBase):
    name: Optional[str] = None
    duration_minutes: Optional[int] = None
    price: Optional[Decimal] = None


class ServiceResponse(ServiceBase):
    id: str
    user_id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True